src/app/api/v1/slack/rotation/on-holiday-post/route.ts

```diff
@@ -0,0 +1,57 @@
+import { PostgresClient } from 'utils/database';
+import { NextRequest, NextResponse } from 'next/server';
+import {
+    getSlackMessage,
+    SlackResponseType,
+    sanitizeSlackText,
+    parsePayloadFromRequest,
+} from 'utils/slack';
+
+export async function POST(req: NextRequest) {
+    try {
+        const parsedPayload = await parsePayloadFromRequest(req);
+        const { text, team_domain } = parsedPayload;
+
+        // Extract and sanitize Slack ID, convert to uppercase, else it breaks
+        const slackIdMatch = text.match(/<@([A-Z0-9]+)>/);
+        const slackId = slackIdMatch ? sanitizeSlackText(slackIdMatch[1]).toUpperCase() : null;
+
+        const rotationName = sanitizeSlackText(text.replace(/<@[^>]+>/g, '').trim());
+
+        if (!rotationName) {
+            return NextResponse.json({ error: 'rotationName is required' }, { status: 400 });
+        }
+
+        if (!slackId) {
+            return NextResponse.json({ error: 'Slack user ID is required' }, { status: 400 });
+        }
```

Comment:
Let's use this as an opportunity to educate the users with the right commands, let's send a SlackMessage that says what's wrong, and then throws an usage example

Reply:
It's a good approach but it will be better if we assign a command to it like /rr-holiday. Right now uses the same base as the command for rotation, so even if we are trying to use a rotation the slack user id is required message will populate and the case were no rotation name is used it will appear 2 times

Reply:
that's already implemented.